src/core/crypto/crypto_platform.cpp

```diff
@@ -658,6 +658,18 @@ OT_TOOL_WEAK otError otPlatCryptoEcdsaVerify(const otPlatCryptoEcdsaPublicKey *a
     return error;
 }
 
+OT_TOOL_WEAK otError otPlatCryptoImportKeyPair(const otPlatCryptoEcdsaKeyPair *aKeyPair,
+                                               uint8_t                        *aDerBytes,
+                                               uint8_t                         aDerLength)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(aDerLength <= OT_CRYPTO_ECDSA_MAX_DER_SIZE, error = kErrorFailed);
+    memcpy(aKeyPair->mDerBytes, aDerBytes, aDerLength);
```

Comment:
`aKeyPair` is defined as `const` (`const otPlatCryptoEcdsaKeyPair *aKeyPair`). 
-  A user of API won't expect anything in `aKeyPair` to change when calling this function.
- But we do copy and update bytes at`mDerBytes`?

I know we can still access a pointer into a `const` struct and copy bytes into the `mDerBytes` (C/C++ allows this since the `mDerBytes` itself is not declared as `const`, i.e pointer is `const` but not what it is pointing to). In general I would advice not using this model (of accessing pointer bytes within a `const` struct). It is not a good coding style.